### PR TITLE
Release notes for Keycloak 24 with OIDC contributions

### LINKED ISSUE #25729: Update release notes for Keycloak 24
### Description

This task is just about not forget to update release notes for various improvements related to the core team: https://github.com/keycloak/keycloak/issues?q=is%3Aissue+is%3Aclosed+label%3Arelease%2F24.0.0+label%3Arelease-notes+label%3Ateam%2Fcore+ and https://github.com/keycloak/keycloak/issues?q=is%3Aissue+is%3Aclosed+label%3Arelease%2F24.0.0+label%3Arelease-notes+label%3Ateam%2Fcore-clients

NOTE: For EdDSA support, the contributors are Takashi and Muhammad Zakwan Bin Mohd Zahid - https://github.com/MuhammadZakwan

### PR DESCRIPTION
**Pull request by mposolda**
closes #25729

### CODE REVIEW

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 1]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
 
 The Keycloak JS adapter now sets the `pkceMethod` option to `S256` by default. This enables Proof Key Code Exchange (https://datatracker.ietf.org/doc/html/rfc7636[PKCE]) for all applications using the adapter. If you are using the adapter on a system that doesn't support PKCE, you can set the `pkceMethod` option to `false` to disable it.
 
+= OAuth/OIDC related improvements
+
+== Lightweight access tokens support
+
+This release contains support for Lightweight access tokens. This allows to use smaller access tokens for specified clients. Such tokens will not contain much
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:110]:
```suggestion
This release contains support for Lightweight access tokens. As a result, you can have smaller access tokens for specified clients. These tokens will have only a few 
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 2]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+claims and hence will be much shorter in size. Note that lightweight access token is still JWT signed by the realm key by default and still contains some very basic claims.
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:111]:
```suggestion
claims, resulting in a smaller size. Note that lightweight access token is still JWT signed by the realm key by default and still contains some very basic claims.
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 3]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+
+From this release, there is a flag *Add to lightweight access token* available on some OIDC protocol mappers. This flag allows to specify if particular claim will be added to lightweight
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:113]:
```suggestion
This release introduces an*Add to lightweight access token*  flag that is available on some OIDC protocol mappers. Use this flag to specify if a particular claim should be added to a lightweight
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 4]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+access token. It is *OFF* by default, so most of the claims are not added by default. There is also client policy executor, which allows to specify if particular client request
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:114]:
```suggestion
access token. It is *OFF* by default, which means that most claims are not added. Also, a client policy executor exists. Use it to specify if a particular client request
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 5]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+should use lightweight access tokens or regular access tokens. Note that in previous release, there was already added switch *Add to token introspection*, so it is possible to add
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:115]:
```suggestion
should use lightweight access tokens or regular access tokens. A previous release added  *Add to token introspection* switch. You use it to add
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 6]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+claims, which are not present in the access token, into introspection endpoint response. Thanks to https://github.com/skabano[Shigeyuki Kabano] for the contribution and Thanks to
```
[the remark this answers is not in the document]
Comment (andymunro) [reply docs/documentation/release_notes/topics/24_0_0.adoc:116]: Use "that" to modify the meaning of the previous clause.
Use "which" to provide an additional clause that does not affect the meaning of the previous clause.
Also, precede "which" with a comma, but do not precede "that" with a comma.

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 7]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+https://github.com/tnorimat[Takashi Norimatsu] for a help and review of this feature.
+
+== Scope parameter supported in the refresh token flow
+
+Since this release, we support `scope` parameter in the OAuth2/OIDC endpoint for token refresh. Note that it is possible to use it to request access tokens with smaller amount
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:121]:
```suggestion
Starting with this release, we support  the`scope` parameter in the OAuth2/OIDC endpoint for token refresh. You can use this parameter to request access tokens with a smaller amount
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 8]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+of scopes than originally granted (so it is not possible to increase access token scope) and it does not affect scope of the refreshed refresh token. This works as
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:122]:
```suggestion
of scopes than originally granted, which means you cannot increase access token scope, and it does not affect the scope of the refreshed refresh token. This function works as
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 9]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+described in the OAuth2 specification. Thanks to https://github.com/cgeorgilakis[Konstantinos Georgilakis] for the contribution.
+
+== Client policy executor for enforcing DPoP
+
+There is new client policy executor `dpop-bind-enforcer`, which can be used to enforce DPoP is used for particular client. This is available just if `dpop` preview
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:127]:
```suggestion
A new client policy executor `dpop-bind-enforcer` is introduced. You can use it to enforce DPoP for a particular client if `dpop` preview
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 10]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+feature is enabled. Thanks to https://github.com/tnorimat[Takashi Norimatsu] for the contribution.
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:128]:
```suggestion
 is enabled. Thanks to https://github.com/tnorimat[Takashi Norimatsu] for the contribution.
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 11]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+
+== Supporting EdDSA
+
+It is possible to create EdDSA realm keys and use them as signature algorithms for various clients. For instance to sign tokens or for client authentication with signed JWT.
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:132]:
```suggestion
You can create EdDSA realm keys and use them as signature algorithms for various clients. For instance, you can use these keys to sign tokens or for client authentication with signed JWT.
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 12]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+This includes also identity brokering where Keycloak itself signs client assertion used for `private_key_jwt` authentication to 3rd party identity provider. Thanks to
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:133]:
```suggestion
This feature includes identity brokering where {project_name} itself signs client assertions that are used for `private_key_jwt` authentication to third party identity provider. Thanks to
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 13]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/MuhammadZakwan[Muhammad Zakwan Bin Mohd Zahid] for the contribution.
+
+== Option to add X509 thumbprint to JWT when using private_key_jwt authentication for identity providers
+
+OIDC identity providers now have the option *Add X.509 Headers to the JWT* for the case when client authentication with JWT signed by private key is used. This can be useful
+for interoperability with some identity providers like Azure AD, which require the thumbprint to be present on the JWT. Thanks to https://github.com/MikeTangoEcho[MT] for the contribution.
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:139]:
```suggestion
OIDC identity providers now have the option *Add X.509 Headers to the JWT* for the case when client authentication with JWT signed by private key is used. This option can be useful
for interoperability with some identity providers such as Azure AD, which require the thumbprint to be present on the JWT. Thanks to https://github.com/MikeTangoEcho[MT] for the contribution.
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 14]
```diff
@@ -103,6 +103,46 @@ The Keycloak JS adapter now uses the https://webpack.js.org/guides/package-expor
...
+
+= Cors improvements
+
+The CORS related Keycloak functionality was extracted into the SPI, which can allow additional flexibility. Just a note that `CorsSPI` is internal and may change in the
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:143]:
```suggestion
The CORS related Keycloak functionality was extracted into the SPI, which can allow additional flexibility. Note that `CorsSPI` is internal and may change in the
```

**Review by andymunro — CHANGES_REQUESTED** [review]
Looks good, @mposolda.  A few changes to suggest.

**Review by tnorimat — APPROVED** [review]

**Comment by mposolda** [comment]
@andymunro Applied all your suggestions. Thanks!
@tnorimat Also Thanks for the review!

**Pull request merged**